File: robocode-tankroyale-bot-api-dotnet-core/events/BulletHitBulletEvent.cs
namespace Robocode.TankRoyale
{
  /// <summary>
  /// Event occurring when a bullet has collided with another bullet.
  /// </summary>
  public sealed class BulletHitBulletEvent : Event
  {
    /// <summary>Bullet that hit another bullet.<summary>
    BulletState Bullet { get; }

    /// <summary>The other bullet that was hit by the bullet.<summary>
    BulletState HitBullet { get; }

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="bullet">Bullet that hit another bullet.</param>
    /// <param name="hitBullet">The other bullet that was hit by the bullet.</param>
    /// <returns></returns>
    public BulletHitBulletEvent(int turnNumber, BulletState bullet, BulletState hitBullet) : base(turnNumber) =>
      (Bullet, HitBullet) = (bullet, hitBullet);
  }
}

File: bot-api/python/src/robocode_tank_royale/bot_api/events/bullet_hit_bullet_event.py
from dataclasses import dataclass

from ..bullet_state import BulletState
from .bot_event import BotEvent


@dataclass(frozen=True, repr=True)
class BulletHitBulletEvent(BotEvent):
    """
    Event triggered when a bullet collides with another bullet in the arena.

    Attributes:
        bullet (BulletState): The bullet that collided with another bullet.
        hit_bullet (BulletState): The bullet that was hit during the collision.
    """
    bullet: BulletState
    hit_bullet: BulletState
